tests/plugins/widget/undo.js

```diff
@@ -448,6 +448,24 @@
 				assert.areSame( 0, objToArray( editor.widgets.instances ).length, '0 widgets after redo' );
 				assertCommands( editor, true, false, 'after redo' );
 			} );
+		},
+
+		// (#3245)
+		'test undo steps': function() {
+			var editor = this.editor;
+
+			this.editorBot.setData( widgetData1, function() {
+				var widget = getWidgetById( editor, 'w1' );
+
+				editor.resetUndo();
+				editor.fire( 'blur' );
+
+				widget.focus();
+
+				editor.fire( 'saveSnapshot' );
+
+				assert.areEqual( 2, editor.undoManager.snapshots.length, 'UndoManager should have one snapshot' );
```

Comment:
This assert have expected value of `2` but the message says `UndoManager should have one snapshot`? And without the fix the actual value is `1` (and I was expecting it to be 3 - one additional undo step). ARe you sure it check the fix correctly?

Reply:
Yup, something is wrong with this assertion, I'll rework this TC.